account_invoice_start_end_dates_tax_change/wizards/account_move_apply_tax_change.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+# NOTE: method copied from recent Odoo versions (odoo.tools.date_utils)
+def date_subtract(value, *args, **kwargs):
+    """
+    Return the difference between ``value`` and a :class:`relativedelta`.
+
+    :param value: initial date or datetime.
+    :param args: positional args to pass directly to :class:`relativedelta`.
+    :param kwargs: keyword args to pass directly to :class:`relativedelta`.
+    :return: the resulting date/datetime.
+    """
+    return value - relativedelta(*args, **kwargs)
+
+
+class AccountMoveApplyTaxChange(models.TransientModel):
+    _inherit = "account.move.apply.tax.change"
+
+    def _skip_line(self, line):
+        if line.start_date or line.end_date:
```

Comment:
```suggestion
        if line.start_date and line.end_date:
```
Otherwise we'll probably get a TypeError comparing Date and Boolean types.

Reply:
I did it this way as the start/end dates module already make sure to have both date fields filled up if one of them is set. But yes, better with an `and` regarding the code below.